### ISSUE TITLE
apiVersion 'v2' is not valid. The value must be "v1" 

### ISSUE THREAD

**Issue by zeph** [argoproj/argo-cd#3901]
```
$ argocd app create -f argocd-dst.yaml 
FATA[0002] rpc error: code = InvalidArgument desc = application spec is invalid: InvalidSpecError: Unable to get app details: rpc error: code = Unknown desc = `helm2 inspect values .` failed exit status 1: Error: error unpacking vault in hvault: apiVersion 'v2' is not valid. The value must be "v1" 

$ grep -r v2 *
charts/vault/.circleci/config.yml:            GO111MODULE=on go get github.com/mikefarah/yq/v2
charts/vault/Chart.yaml:apiVersion: v2
```
I got all my details related in a previously closed ticket https://github.com/argoproj/argo-cd/issues/3280#issuecomment-653432026
 (including the complete helm chart... mine, not the subchart, just referenced)

the above is another attempt, checking in forcefully the extracted subchart + commenting out the requirements.yaml

**Comment by zeph** [comment]
I had to forcefully go back to helm Chart 0.3.2 of vault due to this

**Comment by yaron-idan** [comment]
I've just ran into the same issue with [this chart](https://github.com/niclic/helm-charts/tree/master/prometheus-statsd-exporter), any idea how to work around it?

**Comment by guyelia** [comment]
> I've just ran into the same issue with [this chart](https://github.com/niclic/helm-charts/tree/master/prometheus-statsd-exporter), any idea how to work around it?

Same here, the same chart.

**Comment by aekrohn** [comment]
I am also trying to install Hashicorp Vault using ArgoCD. The application has a requirements file pointing to the Vault helm chart repository, and that repo only has the latest version, so an older, working version of the chart and ArgoCD are not compatible for our use case.

Here is the full error given by ArgoCD

```
nable to create application: application spec is invalid: InvalidSpecError: Unable to generate manifests in vault: rpc error: code = Unknown desc = `helm2 template . --name hashicorp-vault --namespace argocd --kube-version 1.19 --values values.yml --api-versions v1 --api-versions apiregistration.k8s.io/v1 --api-versions apiregistration.k8s.io/v1beta1 --api-versions extensions/v1beta1 --api-versions apps/v1 --api-versions events.k8s.io/v1 --api-versions events.k8s.io/v1beta1 --api-versions authentication.k8s.io/v1 --api-versions authentication.k8s.io/v1beta1 --api-versions authorization.k8s.io/v1 --api-versions authorization.k8s.io/v1beta1 --api-versions autoscaling/v1 --api-versions autoscaling/v2beta1 --api-versions autoscaling/v2beta2 --api-versions batch/v1 --api-versions batch/v1beta1 --api-versions certificates.k8s.io/v1 --api-versions certificates.k8s.io/v1beta1 --api-versions networking.k8s.io/v1 --api-versions networking.k8s.io/v1beta1 --api-versions policy/v1beta1 --api-versions rbac.authorization.k8s.io/v1 --api-versions rbac.authorization.k8s.io/v1beta1 --api-versions storage.k8s.io/v1 --api-versions storage.k8s.io/v1beta1 --api-versions admissionregistration.k8s.io/v1 --api-versions admissionregistration.k8s.io/v1beta1 --api-versions apiextensions.k8s.io/v1 --api-versions apiextensions.k8s.io/v1beta1 --api-versions scheduling.k8s.io/v1 --api-versions scheduling.k8s.io/v1beta1 --api-versions coordination.k8s.io/v1 --api-versions coordination.k8s.io/v1beta1 --api-versions node.k8s.io/v1beta1 --api-versions discovery.k8s.io/v1beta1 --api-versions argoproj.io/v1alpha1` failed exit status 1: Error: error unpacking vault-0.7.0.tgz in hashicorp-vault: apiVersion 'v2' is not valid. The value must be "v1"
```

**Comment by leewardbound** [comment]
I have encountered this with the Nextcloud chart v2.2.1 https://github.com/nextcloud/helm/tree/master/charts/nextcloud, but I want to note that it only began when I changed from the "remote-chart" Application model (`application.yaml` has the remote chart URL, with values included under the `helm:` key) to the "depenency-chart" Application model (`application.yaml` points to a directory with `Chart.yaml` and `requirements.yaml` and `values.yaml` file).

I have many other helm charts installed with the dependency chart model, but this one fails no matter what I have tried. The only resolution, for me, is to go back to installing the chart using the "remote-chart" model (which, IMHO, is far less clean).

**Comment by aekrohn** [comment]
I was able to resolve this, and it doesn't appear to be a bug, just a missing parameter on my part. My argocd application points to a repo directory that contains `Chart.yaml`, `requirements.yaml`, and `values.yaml`. The `requirements.yaml` file is what actually contains the Hashicorp repo information:

```
dependencies:
- name: vault
  version: v0.7.0
  repository: https://helm.releases.hashicorp.com
```
And `Chart.yaml` lets argocd know we're doing a helm chart:

```
name: my-project-vault
version: v0.0.1
```
All I needed to do was add `apiVersion: v2` to `Chart.yaml` and it started working.
